backend/api/badge_scanner/schema.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import pretix
+import re
+from typing import Any
+import strawberry
+from strawberry.types.info import Info
+
+from api.permissions import IsAuthenticated
+from badge_scanner import models
+from conferences.models import Conference
+from users.client import get_user_by_email
+
+
+@strawberry.type
+class ScanError:
+    message: str
+
+
+@strawberry.input
+class ScanBadgeInput:
+    url: str
+    conference_code: str
+
+    url_regex = re.compile(r"^https://pycon\.it/b/([\w\d-]+)$")
+
+    def validate(self) -> ScanError | None:
+        if not self.url:
+            return ScanError(message="URL is required")
+
+        match = self.url_regex.match(self.url)
+
+        return None if match else ScanError(message="URL is not valid")
+
+    @property
+    def order_position_id(self) -> str:
+        match = self.url_regex.match(self.url)
+
+        assert match
+
+        return match.group(1)
```

Comment:
This should be changed to decode the ID as well as we won't have the raw ID in the URL

e.g it won't be `pycon.it/b/132` but `pycon.it/b/abc`

We should do something similar to what we did for submissions, use hashids and the `decode_hashid` and `encode_hashid` functions (it should be fine to still use the secret key as salt, we can think of improving it later)